builtin/game/misc.lua

```diff
@@ -4,32 +4,69 @@
 -- Misc. API functions
 --
 
-core.timers_to_add = {}
-core.timers = {}
-core.register_globalstep(function(dtime)
-	for _, timer in ipairs(core.timers_to_add) do
-		table.insert(core.timers, timer)
-	end
-	core.timers_to_add = {}
-	local index = 1
-	while index <= #core.timers do
-		local timer = core.timers[index]
-		timer.time = timer.time - dtime
+local timers_to_add = {}
+local timers = {}
+local mintime
+local function update_timers(delay)
+	mintime = false
+	local sub = 0
+	for index = 1,#timers do
+		index = index - sub
+		local timer = timers[index]
+		timer.time = timer.time - delay
 		if timer.time <= 0 then
 			timer.func(unpack(timer.args or {}))
-			table.remove(core.timers,index)
+			table.remove(timers, index)
+			sub = sub + 1
+		elseif mintime
+		and mintime ~= true then
+			mintime = math.min(mintime, timer.time)
+		else
+			mintime = timer.time
+		end
+	end
+end
+
+local function update_timers_to_add(delay)
```

Comment:
>Why do you call update_timers(delay) twice?

It's needed because delay is subtracted from the timers.time in the update_timers function.
And delay needs to be set to 0 else the fresh timers.time loose delay time although they didn't exist at the time delay was increasing.